proto/p4/v1/p4runtime.proto

```diff
@@ -834,7 +834,10 @@ enum SdnPort {
 }
 
 //------------------------------------------------------------------------------
-message CapabilitiesRequest {}
+message CapabilitiesRequest {
+  // Added in v1.5.0
+  uint64 device_id = 1;
```

Comment:
Is there a way to have this be optional?
The `p4runtime_api_version` would typically be the same for all devices controlled by a P4Runtime server, so if a client just wants to query that information, now it has to provide a `device_id` for no reason.
The notion of capabilities is (intentionally?) pretty broad, but I expect some would be device-independent, and some may be device-specific. Could `device_id` be defined as an optional filter to limit the capabilities in the response to the ones applicable to that specific device? If no `device_id` is provided, the response should only include capabilities which are applicable across all devices, or should include a mapping from device ID to device-specific capabilities in the `experimental` field.

Reply:
That seems like a reasonable approach to me. Let's say "If unset (or 0), should return only capabilities common to all devices controlled by the server. Otherwise, should return any capabilities applicable to the given device." or something like that?